nanoFramework.Device.Bluetooth/GenericAttributeProfile/GattServiceProvider.cs

```diff
@@ -96,30 +93,85 @@ public void StartAdvertising()
         public void StartAdvertising(GattServiceProviderAdvertisingParameters parameters)
         {
             // Check and switch to server mode
-            BluetoothNanoDevice.CheckMode(BluetoothNanoDevice.Mode.Server);
-
-            BluetoothNanoDevice.DeviceName = parameters.DeviceName;
+            if (BluetoothNanoDevice.RunMode != BluetoothNanoDevice.Mode.Server )
+            {
+                BluetoothLEServer.Instance.Start();
+            }
 
             // Save parameters
             _isConnectable = parameters.IsConnectable;
             _isDiscoverable = parameters.IsDiscoverable;
             _serviceData = parameters.ServiceData;
 
-            _deviceName = Encoding.UTF8.GetBytes(parameters.DeviceName);
+            // Make sure pairing attributes set
+            BluetoothLEServer.Instance.Pairing.SetPairAttributes();
 
             // Start advertising.
             // Native code will use the data provided by this GattServiceProvider instance to
             // initialise the BLE advert, service and characteristic definitions.
-            if (NativeStartAdvertising())
+            if (NativeStartAdvertising(BluetoothLEServer._services))
             {
                 _status = GattServiceProviderAdvertisementStatus.Started;
             }
         }
 
+        #region Security
+
         /// <summary>
-        ///  Stop advertising the GATT service.
+        /// Evaulate all services and work out default security requirements.
```

Comment:
```suggestion
        /// Evaluates all services and work out default security requirements.
```

Reply:
This method has been removed as not used. New method added in BluetoothLeServer called UpdateSecurityPairingRequirements which is called when Characteristic or Descriptor added.